omero/annotation_scripts/KeyVal_from_csv.py

```diff
@@ -158,28 +159,30 @@ def keyval_from_csv(conn, script_params):
         # Needs omero-py 5.9.1 or later
         temp_name = temp_file.name
         with open(temp_name, 'rt', encoding='utf-8-sig') as file_handle:
+            file_length = len(file_handle.read(-1))
```

Comment:
Instead of reading the whole file to get the length, you can get this from the original_file object above (before you open the file)..
```
file_length = original_file.size.val
```

Reply:
Yes, when I wrote this I was almost sure that a more elegant solution would exist.
Thanks!